src/main/java/org/gradle/profiler/studio/launcher/LauncherConfigurationParser.java

```diff
@@ -95,57 +50,37 @@ public LauncherConfigurationParser withStudioAgentParameters(int studioAgentPort
+        Path startCommand = getStartCommand(studioInstallDir);
+        List<String> additionalJvmArgs = getAdditionalJvmArgs();
 
-        Path agentJar = GradleInstrumentation.unpackPlugin("studio-agent").toPath();
-        Path asmJar = GradleInstrumentation.unpackPlugin("asm").toPath();
-        Path supportJar = GradleInstrumentation.unpackPlugin("instrumentation-support").toPath();
-        Path protocolJar = GradleInstrumentation.unpackPlugin("client-protocol").toPath();
-        Path studioPlugin = GradleInstrumentation.unpackPlugin("studio-plugin").toPath();
-        Path studioPluginsDir = studioSandbox.getPluginsDir();
-        Path studioLogsDir = studioSandbox.getLogsDir();
-        List<Path> sharedJars = Arrays.asList(asmJar, protocolJar);
-        List<Path> studioPluginJars = Arrays.asList(studioPlugin, protocolJar);
+        // Note: In headless mode ANDROID_HOME and ANDROID_SDK_ROOT have to be set otherwise
+        // Studio will fail since "missing Android SDK" modal will try to open
+        String headlessCommand = SHOULD_RUN_HEADLESS ? "headless-starter" : "";
 
-        String mainClass = studioConfiguration.getMainClass();
-        List<Path> classpath = studioConfiguration.getClasspath();
-        List<String> commandLine = new ArrayList<>();
-        commandLine.add(javaCommand.toString());
-        commandLine.add("-cp");
-        commandLine.add(Joiner.on(File.pathSeparator).join(classpath));
-        systemProperties.forEach((key, value) -> commandLine.add(String.format("-D%s=%s", key, value)));
-        // Support for IntelliJ with Java17 runtime
-        commandLine.addAll(JAVA17_ADD_OPENS);
+        return new LaunchConfiguration(startCommand, headlessCommand, studioInstallDir, additionalJvmArgs, studioSandbox);
+    }
+
+    private List<String> getAdditionalJvmArgs() {
+       List<String> jvmArgs = new ArrayList<>();
         if (enableStudioAgentParameters) {
-            commandLine.add(String.format("-javaagent:%s=%s,%s", agentJar, studioAgentPort, supportJar));
-            commandLine.add("--add-exports");
-            commandLine.add("java.base/jdk.internal.misc=ALL-UNNAMED");
-            commandLine.add("-Xbootclasspath/a:" + Joiner.on(File.pathSeparator).join(sharedJars));
-        }
-        commandLine.addAll(studioJvmArgs);
-        commandLine.add(mainClass);
-        if (SHOULD_RUN_HEADLESS) {
-            // Note: In headless mode ANDROID_HOME and ANDROID_SDK_ROOT have to be set otherwise
-            // Studio will fail since "missing Android SDK" modal will try to open
-            commandLine.add("headless-starter");
+            Path agentJar = GradleInstrumentation.unpackPlugin("studio-agent").toPath();
+            Path supportJar = GradleInstrumentation.unpackPlugin("instrumentation-support").toPath();
+            Path asmJar = GradleInstrumentation.unpackPlugin("asm").toPath();
+            Path protocolJar = GradleInstrumentation.unpackPlugin("client-protocol").toPath();
```

Comment:
❔ Why convert these from `File` to `Path` only to immediately convert them to `String`? IIUC, converting from `File` to `String` would produce the same results, no?

Reply:
Yeah, makes no sense. Will change that.